pygments/lexers/arturo.py

```diff
@@ -0,0 +1,312 @@
+             r'|-->|<->|<-->'
+             r'|<\\|\<|\>'
+             r'|\=\||\|\='
+             r'|\./|\^|\.\.|\.'
+             r'|\\',
+            Operator)
+
+        ],
+            'punctuation': [
+                (r'[()[\],]',   Punctuation),
+            ],
+            'sugar':[
+                (r'->'
+                 r'|=>'
+                 r'|\|'
+                 r'|::',
+                Name.Decorator),
+            ],
+
+        'constants': [
+            include('boolean'),
+            include('character'),
+            include('color'),
+            include('float'),
+            include('integer'),
+            include('label'),
+            include('literal'),
+            include('strings'),
+            include('type'),
+        ],
+
+            'boolean': [
+                (words(('false', 'true', 'maybe'), suffix=r'\b'),
+                    Name.Constant)
+            ],
+            'character': [
+                (r'`.`', String.Char)
+            ],
+            'color': [
+                (r'#\w+',
+                        Name.Constant)
+            ],
+            'float': [
+                (r'[0-9]+\.[0-9]+',
+                        Number.Float)
+            ],
+            'integer': [
+                (r'[0-9]+',
+                        Number.Integer)
+            ],
+            'label': [
+                (r'\w+\b\??:',
+                        Name.Label)
+            ],
+            'literal': [
+                (r'\'(?:\w+\b\??:?)',
+                        Keyword.Declaration)
+            ],
+            'type': [
+                (r'\:\w+',
+                        Keyword.Type)
+            ],
+            'attributes': [
+                (r'\.\w+',
+                        Name.Attribute)
+            ],
+
+
+        'strings': [
+            # Single Line Strings
+            (r'"',   String.Double, 'inside-simple-string'),
+            (r'»',   String.Single, 'inside-smart-string' ),
+            (r'«««', String.Double, 'inside-safe-string'  ),
+            (r'\{\/', String.Single, 'inside-regex-string'),
+
+            # Multi Line Strings
+            (r'\{\:',   String.Double, 'inside-curly-verb-string'),
+            (r'(\{)(\!)(\w+)(\s|\n)([\w\W]*?)(^\})',
+                                         handle_annotated_strings),
+            (r'\{',     String.Single, 'inside-curly-string'     ),
+            (r'\-{3,}', String.Single, 'inside-eof-string'       ),
+        ],
+            'string-basics': [
+                include('string-interpol'),
+                include('string-escape'),
+                include('string-templates'),
+            ],
+                'string-interpol': [
+                    (r'\|',
+                        String.Interpol, 'inside-interpol'), # Interpolation
+                ],
+                    'inside-interpol': [
+                        (r'\|', String.Interpol, '#pop'),
+                        (r'.', using(this)),
```

Comment:
this won't work propery, since it will pass single characters to the `this` instance.

rule should be `r'[^|]+'`

Reply:
Wow, that explains a lot to me!

Reply:
- [x] [fix interpolation rule](https://github.com/pygments/pygments/pull/2259/commits/2c08d2242c2145909ebd602a76dee0a8955bf33a)